paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -1305,6 +1321,76 @@ def diagnose_why_instance_is_stuck(
             print(f"    {line}")
     print("")
 
+    if should_ping_for_unhealthy_pods:
+        maybe_ping_for_unhealthy_pods(
+            service, instance, cluster, git_sha, status, notify_fn
+        )
+
+
+already_pinged = False
+
+
+def maybe_ping_for_unhealthy_pods(
+    service: str,
+    instance: str,
+    cluster: str,
+    git_sha: str,
+    status: InstanceStatusKubernetesV2,
+    notify_fn: Optional[Callable[[str], None]] = None,
+) -> None:
+    if notify_fn and not already_pinged:
```

Comment:
nit: I think this check should be done by the caller, not inside the function.

Reply:
I pulled the `notify_fn` check up to the caller.